src/lib/commands.js

```diff
@@ -0,0 +1,317 @@
+		} else if (store.vendorList && Object.keys(store.vendorList).length) {
+			vendorList = JSON.parse(JSON.stringify(store.vendorList));
+			callback(vendorList);
+		} else {
+			const { getVendorList } = config;
+
+			if (getVendorList) {
+				getVendorList((data, err) => {
+					if (err) {
+						callback({}, err);
+					} else {
+						vendorList = data;
+						callback(vendorList);
+					}
+				});
+			}
+		}
+	};
+
+	const vectorToObject = (set) => {
+		let obj = {};
+
+		set.forEach((hasConsent, id) => {
+			obj[id] = hasConsent;
+		});
+
+		return obj;
+	};
+
+	 /**
+	 * Build consent fields object v1 from data that has already been persisted.
+	 */
+	const getConsentFieldsObject = () => {
+		const {
+			persistedConsentData
+		} = store;
+
+		const {
+			created,
+			lastUpdated,
+			cmpId_: cmpId = store.tcModel.cmpId,
+			cmpVersion_: cmpVersion = 1,
+			consentScreen_: consentScreen = store.tcModel.consentScreen,
+			consentLanguage_: consentLanguage = store.tcModel.consentLanguage,
+			vendorListVersion_: vendorListVersion,
+		} = persistedConsentData;
+
+		return {
+			cmpId,
+			cmpVersion,
+			consentLanguage,
+			consentScreen,
+			created,
+			vendorListVersion,
+			lastUpdated,
+		};
+	};
+
+	const generateConsentStringV1 = (data = {}, callback) => {
+		const {
+			persistedConsentData,
+		} = store;
+
+		const {
+			created,
+			lastUpdated,
+			consentLanguage_: consentLanguage,
+			consentScreen_: consentScreen,
+			version_: version = COOKIE_VERSION_V1,
+			cmpId_: cmpId = CMP_ID
+		} = persistedConsentData;
+
+		const {
+			selectedVendorIds = [],
+			selectedPurposeIds = [],
+			maxVendorId,
+			cookieVersion,
+			vendorListVersion,
+			cmpVersion
+		} = data;
+
+		const consentData = {
+			cmpId,
+			maxVendorId,
+			cookieVersion,
+			cmpVersion,
+			vendorListVersion,
+			created,
+			lastUpdated,
+			consentLanguage,
+			consentScreen,
+			version,
+			selectedVendorIds: new Set(arrayFrom(selectedVendorIds)),
+			selectedPurposeIds: new Set(arrayFrom(selectedPurposeIds))
+		};
+
+		const valid = (data = {}) => [
+			'cmpId', 'cmpVersion', 'consentLanguage', 'consentScreen', 'cookieVersion', 'created', 'lastUpdated',
+			'maxVendorId', 'selectedPurposeIds', 'selectedVendorIds', 'vendorListVersion'
+		].every(prop => data.hasOwnProperty(prop));
+
+		callback(valid(consentData) ? encodeVendorConsentData(consentData) : undefined);
+	};
+
+	const getConsentFieldsV1 = (callback, params) => {
+		const data = getConsentFieldsObject();
+
+		generateConsentStringV1({
+			...data,
+			...params
+		}, (metadata) => {
+			const consent = {
+				metadata,
+				gdprApplies: config.gdprApplies,
+				hasGlobalScope: config.storeConsentGlobally,
+				...data,
+				vendorListVersion: params.vendorListVersion
+			};
+
+			callback(consent, true);
+		});
+	};
+
+	const getConsentObject = (callback, params) => {
+		getVendorList((vendorList, err) => {
+			if (err) {
+				callback({}, false);
+			}
+			if (vendorList) {
```

Comment:
Note that it is possible that `err` and `vendorList` may be not empty at the same time. For example empty object casts on `true` value. So please add `return` statement inside previous `if` or use `else if` here instead simple `if`.

Reply:
We do not need getVendorList function here anymore.